torchsynth/module.py

```diff
@@ -303,24 +303,29 @@ def _ramp(self, start, duration: T, inverse: bool = False) -> Signal:
         # Shape ramps.
         ramp = ramp - start_[:, None]
         ramp = torch.maximum(ramp, T(0.0))
-        ramp = ramp / duration_[:, None]
+        ramp = (ramp + EPSILON) / (duration_[:, None] + EPSILON)
         ramp = torch.minimum(ramp, T(1.0))
 
         if inverse:
-            ramp = 1 - ramp
+            ramp = 1.0 - ramp
+            ramp[torch.sum(ramp, axis=1) == 0] = 1.0
```

Comment:
This is a funny workaround. In inverse mode, a ramp with 0 duration (that is all 1's) becomes all 0's, which is a problem for the ultimate calculation of the adsr signal (a * d * r => 0's). So this replaces only rows who sum to 0 (i.e., all components are zero).